fvm/state/transaction_state_test.go

```diff
@@ -526,3 +526,85 @@ func TestFinalizeMainTransaction(t *testing.T) {
 	_, err = txn.Get(registerId)
 	require.ErrorContains(t, err, "cannot Get on a finalized state")
 }
+
+func TestInterimReadSet(t *testing.T) {
+	txn := newTestTransactionState()
+
+	// Test setup
+
+	readRegisterId1 := flow.NewRegisterID("read", "1")
+	readRegisterId2 := flow.NewRegisterID("read", "2")
+	readRegisterId3 := flow.NewRegisterID("read", "3")
+	readRegisterId4 := flow.NewRegisterID("read", "4")
+
+	writeRegisterId1 := flow.NewRegisterID("write", "1")
+	writeValue1 := flow.RegisterValue([]byte("value1"))
+
+	writeRegisterId2 := flow.NewRegisterID("write", "2")
+	writeValue2 := flow.RegisterValue([]byte("value2"))
+
+	writeRegisterId3 := flow.NewRegisterID("write", "3")
+	writeValue3 := flow.RegisterValue([]byte("value3"))
+
+	err := txn.Set(writeRegisterId1, writeValue1)
+	require.NoError(t, err)
+
+	_, err = txn.Get(readRegisterId1)
+	require.NoError(t, err)
+
+	_, err = txn.Get(readRegisterId2)
+	require.NoError(t, err)
+
+	value, err := txn.Get(writeRegisterId1)
+	require.NoError(t, err)
+	require.Equal(t, writeValue1, value)
+
+	_, err = txn.BeginNestedTransaction()
```

Comment:
could you add some comments to explain the behavior that you are testing?

such as why creating a nested transaction here in the tests